app/models/geoname.rb

```diff
@@ -1,75 +1,112 @@
+    analysis: {
+      filter: {
+        synonym: {
+          type: 'synonym',
+          synonyms: SYNONYMS
+        }
+      },
+      analyzer: {
+        custom_analyzer: {
+          type: 'custom',
+          tokenizer: 'whitespace',
+          filter: %w(standard lowercase synonym)
+        }
+      }
+    }
+  }
 
-  SYNONYMS = ["ft, fort", "st, saint", "afb, air force base", "afs, air force station", "ang, air national guard", "junc, junction", "spgs, springs", "natl, nat, national", "pk, park", "newcastle, new castle", "cavecreek, cave creek"]
+  settings index: SETTINGS do
+    mappings dynamic: 'false' do
+      indexes :type, type: 'keyword'
+      indexes :location, type: 'text', analyzer: 'custom_analyzer'
+      indexes :state, type: 'text', analyzer: 'keyword'
+      indexes :geo, type: 'geo_point'
+      indexes :id, type: 'keyword', index: false
+    end
+  end
 
   class << self
 
+    def client
+      @client ||= Geoname.__elasticsearch__.client
+    end
+
     def create_search_index
-      Tire.index index_name do
-        create(
-          settings: {
-            index: {
-              analysis: {
-                analyzer: {custom_analyzer: {type: 'custom', tokenizer: 'whitespace', filter: %w(standard lowercase synonym)}},
-                filter: {synonym: {type: 'synonym', synonyms: SYNONYMS}}
-              }
-            }
-          },
-          mappings: {
-            geoname: {
-              properties: {
-                type: {type: 'string'},
-                location: {type: 'string', analyzer: 'custom_analyzer'},
-                state: {type: 'string', analyzer: 'keyword'},
-                geo: {type: 'geo_point'},
-                id: {type: 'string', index: :not_analyzed, include_in_all: false}
-              }
-            }
-          }
-        )
-      end
+      client.indices.create(
+        index: INDEX_NAME,
+        body: { settings: settings.to_hash, mappings: mappings.to_hash }
+      )
     end
 
     def geocode(options = {})
-      search_for(options.merge(size: 1)).results.first.geo.to_hash rescue nil
+      search_for(options.merge(size: 1)).results.first.geo rescue nil
     end
 
     def search_for(options)
-      Tire.search index_name do
-        query do
-          boolean do
-            must { match :location, options[:location], operator: 'AND' }
-            must { term :state, options[:state] }
-          end
-        end
-        size options[:size]
-      end
+      search_definition = {
+        query: {
+          bool: {
+            must: [
+              { match: { location: { query: options[:location], operator: 'AND' } } },
+              { term: { state: options[:state] } }
+            ]
+          }
+        }
+      }
+
+      search_definition[:size] = options[:size]
+
+      Geoname.search(search_definition, index: INDEX_NAME)
     end
 
     def delete_search_index
-      search_index.delete
+      client.indices.delete index: INDEX_NAME rescue nil
```

Comment:
I think you're rescuing `nil` here because `client.indices` might not exist for `INDEX_NAME`. In that case could you either check first (`client.indices.delete(index: INDEX_NAME) if search_index_exists?`) or rescue a more specific error here? I don't want to drop other exceptions on the floor unnecessarily.